src/main/java/com/exasol/bucketfs/list/ListingProvider.java

```diff
@@ -0,0 +1,91 @@
+package com.exasol.bucketfs.list;
+
+import static com.exasol.bucketfs.BucketConstants.PATH_SEPARATOR;
+import static com.exasol.bucketfs.BucketOperation.LIST;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.*;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+import com.exasol.bucketfs.*;
+
+/**
+ * This class enables to request list of buckets or of objects inside a bucket.
+ */
+public abstract class ListingProvider {
```

Comment:
```suggestion
public abstract class AbstractListingProvider {
```

Rename other instance accordingly, please.